### PR TITLE
Feature/get request

### LINKED ISSUE #301: Add GetRequest method to BaseRequest
In order to use the strongly typed builders to create a native request we need a GetRequest() method.  This is useful for constructing requests for Batch or for other scenarios where we want to make the call manually to process the response differently.

### PR DESCRIPTION
**Pull request by baywet**
Fixes #301
Depends on core 1.0.3
Depends on https://github.com/microsoftgraph/msgraph-sdk-java-core/pull/70
Depends on https://github.com/microsoftgraph/msgraph-sdk-java-core/issues/71

### Changes proposed in this pull request
-

### Other links

-
TODO: once this PR gets merged, ping the graph explorer API repo to update the sample https://docs.microsoft.com/en-us/graph/sdks/batch-requests?context=graph%2Fapi%2F1.0&view=graph-rest-1.0&tabs=java

### CODE REVIEW

**Comment by zengin** [comment]
@baywet, can you please summarize which parts of the code are generated, which parts are hand-crafted?

**Comment by baywet** [comment]
@zengin this commit is the only code-gen in the PR https://github.com/microsoftgraph/msgraph-sdk-java/pull/463/commits/aa0b2c3f370b9b477a557a61edfe5bd0a1544d85 the rest is handcrafted.

basically I'm adding 2 methods + 2 overloads for one method (withHttpMethod and getHttpRequest) in IHttpRequest and implementing that in the different BaseRequest classes.

Then I'm relying on IHttpRequest inheritance (the code-gen part) to expose the methods in the object model without having to cast.

**Review by zengin — APPROVED** [review]
LGTM

**Review by MIchaelMainer — COMMENTED** [review]
You get my review post merge. No required actions

File: src/main/java/com/microsoft/graph/http/CoreHttpProvider.java [hunk 1]
```diff
@@ -203,7 +203,137 @@ public void run() {
...
+			if (!hasHeader(requestHeaders, Constants.CONTENT_TYPE_HEADER_NAME)) {
+				corehttpRequestBuilder.addHeader(Constants.CONTENT_TYPE_HEADER_NAME, Constants.JSON_CONTENT_TYPE);
+				contenttype = Constants.JSON_CONTENT_TYPE;
+			}
+		}
+
+		RequestBody requestBody = null;
+		// Handle cases where we've got a body to process.
+		if (bytesToWrite != null) {
+			final String mediaContentType = contenttype;
```
[the remark this answers is not in the document]
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/CoreHttpProvider.java:299]: I agree on the fact that sendRequestInternal was waaay too long to be testable. The fact that this change already breaks it into to pieces is a step in the right direction for better testability. And yes, moving this section to a dedicated method would be a good next step. Same thing for the part that deals with the response in sendRequestInternal

File: src/main/java/com/microsoft/graph/http/IHttpRequest.java [hunk 2]
```diff
@@ -154,5 +158,37 @@
...
+
+    /**
+     * Returns the Request object to be executed
+     * @param serializedObject the object to serialize at the body of the request
+     * @param progress the progress callback
+     * @param <requestBodyType> the type of the serialized object
+     * @param <responseType> the type of the response object
+     * @return the Request object to be executed
+     */
+    <requestBodyType, responseType> Request getHttpRequest(final requestBodyType serializedObject, final IProgressCallback<responseType> progress) throws ClientException;
```
Comment (baywet) [inline src/main/java/com/microsoft/graph/http/IHttpRequest.java:192]: IHttpProvider should not contain an implementation specific type and should be generic instead

**Pull request merged**